internal/worker/caasapplicationprovisioner/ops.go

```diff
@@ -752,6 +761,31 @@ func ensureScale(appName string, app caas.Application, appLife life.Value,
 	return nil
 }
 
+func reconcileApplicationStorage(appName string, app caas.Application, facade CAASProvisionerFacade,
+	lastApplied *[]storage.KubernetesFilesystemParams, logger Logger) error {
+	logger.Debugf("reconciling application %q storage", appName)
+
+	// Get filesystem provisioning info.
+	info, err := facade.FilesystemProvisioningInfo(appName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	// Set its initial value. It must be the first time it was invoked.
+	if *lastApplied == nil {
+		*lastApplied = info.Filesystems
+		logger.Infof("[adis][reconcileappstorage] app: %q first time...", appName)
+		return nil
+	}
+	// TODO(adisazhar123): enumerate lastapplied and filesystems and check the values
+	if reflect.DeepEqual(*lastApplied, info.Filesystems) {
```

Comment:
note here: 

perhaps we dont have to use reflect here since it could be quite expensive? I checked these are the values that are used:

storageName, provider type, size
We used one key for attributes in tmpfs but perhaps all these could be sorted out in a helper func in that package